extractorapp/src/main/java/org/georchestra/extractorapp/ws/extractor/WfsExtractor.java

```diff
@@ -213,9 +228,13 @@ public static void enablePreemptiveBasicAuth(URL capabilitiesURL, HttpClientBuil
      * @return the directory that contains the extracted file
      */
     public File extract (ExtractorLayerRequest request) throws IOException, TransformException, FactoryException {
-        if (request._owsType != OWSType.WFS) {
+    	checkNotNull(request);
+    	if (request._owsType != OWSType.WFS) {
```

Comment:
https://github.com/georchestra/georchestra/pull/2582/commits/94bc8f06d32aeb61271f42d3dd01cf41bd9baec0#diff-812f3e1b48408b609afa21a720287c4aR228 tended to unify contract reading.

Reply:
I'm not sure what the problem is here?